packages/logger/src/node.ts

```diff
@@ -0,0 +1,159 @@
+import path from "node:path";
+import DailyRotateFile from "winston-daily-rotate-file";
+import TransportStream from "winston-transport";
+import winston from "winston";
+import {Logger, LogLevel, logLevelNum, TimestampFormat} from "./interface.js";
+import {ConsoleDynamicLevel} from "./utils/consoleTransport.js";
+import {getFormat} from "./utils/format.js";
+import {WinstonLogger} from "./winston.js";
+
+const DATE_PATTERN = "YYYY-MM-DD";
+
+export type LoggerNodeOpts = {
+  level: LogLevel;
+  /**
+   * Enable file output transport if set
+   */
+  file?: {
+    filepath: string;
+    /**
+     * Log level for file output transport
+     */
+    level: LogLevel;
+    /**
+     * Rotation config for file output transport
+     */
+    dailyRotate?: number;
+  };
+  /**
+   * Module prefix for all logs
+   */
+  module?: string;
+  /**
+   * Rendering format for logs, defaults to "human"
+   */
+  format?: "human" | "json";
+  /**
+   * Set specific log levels by module
+   */
+  levelModule?: Record<string, LogLevel>;
+  /**
+   * Enables relative to genesis timestamp format
+   * ```
+   * timestampFormat = {
+   *   format: TimestampFormatCode.EpochSlot,
+   *   genesisTime: args.logFormatGenesisTime,
+   *   secondsPerSlot: config.SECONDS_PER_SLOT,
+   *   slotsPerEpoch: SLOTS_PER_EPOCH,
+   * }
+   * ```
+   */
+  timestampFormat?: TimestampFormat;
+};
+
+export type LoggerNodeChildOpts = {
+  module?: string;
+};
+
+export type LoggerNode = Logger & {
+  toOpts(): LoggerNodeOpts;
+  child(opts: LoggerNodeChildOpts): LoggerNode;
+};
+
+/**
+ * Setup a CLI logger, common for beacon, validator and dev commands
+ */
+export function getNodeLogger(opts: LoggerNodeOpts): LoggerNode {
+  return WinstonLoggerNode.fromNewTransports(opts);
+}
+
+function getNodeLoggerTransports(opts: LoggerNodeOpts): winston.transport[] {
+  const consoleTransport = new ConsoleDynamicLevel({
+    // Set defaultLevel, not level for dynamic level setting of ConsoleDynamicLvevel
+    defaultLevel: opts.level,
+    debugStdout: true,
+    handleExceptions: true,
+  });
+
+  if (opts.levelModule) {
+    for (const [module, level] of Object.entries(opts.levelModule)) {
+      consoleTransport.setModuleLevel(module, level);
+    }
+  }
+
+  const transports: TransportStream[] = [consoleTransport];
+
+  // yargs populates with undefined if just set but with no arg
+  // $ ./bin/lodestar.js beacon --logFileDailyRotate
+  // args = {
+  //   logFileDailyRotate: undefined,
+  // }
+  // `lodestar --logFileDailyRotate` -> enabled daily rotate with default value
+  // `lodestar --logFileDailyRotate 10` -> set daily rotate to custom value 10
+  // `lodestar --logFileDailyRotate 0` -> disable daily rotate and accumulate in same file
+  if (opts.file) {
+    const filename = opts.file.filepath;
+
+    transports.push(
+      opts.file.dailyRotate != null
```

Comment:
What's the expected approach to disable log file rotation now?

The comment above does not longer match behavior as setting `--logFileDailyRotate 0` will still initialize `DailyRotateFile`. Default `winston.transports.File` is effectively never used now.

Reply:
Right, logFileDailyRotate is a bit a weird UX flag. Does yargs support accepting both syntax with no arg `--logFileDailyRotate` and with arg `--logFileDailyRotate 5`? That would make handling easier